### DISCUSSION TITLE
[Contract Verification] Gaslite Drop - can't verify on testnet

### DISCUSSION THREAD

**Discussion by coffee-the-dev** [zkSync-Community-Hub/zksync-developers#756] [Contract Verification] [answered]
### Team or Project

Abstract

### Environment

Testnet

### Block Explorer

https://explorer.testnet.abs.xyz/

### Issue Type

- [ ] Plugin
- [ ] Block Explorer

### Contract Address

0xF8289FD0DC7813babfe5c2948f02D40757b7CFF6

### Compiler Type

Single file

### zkSolc Version

1.5.4

### Solc Version

0.8.19

### Contract Name

GasliteDrop

### Contract Code

https://github.com/PopPunkLLC/GasliteDrop/blob/main/contracts/src/GasliteDrop.sol

### Constructor Arguments

N/A

### Repo Link (Optional)

https://github.com/PopPunkLLC/GasliteDrop

### Additional Details

I deployed and am attempting to verify Gaslite Drop on Abstract Testnet but it will not verify.

Commands used:
```
foundryup-zksync
.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Fork of    : https://github.com/foundry-rs/
Repo       : https://github.com/matter-labs/foundry-zksync/
Book       : https://book.getfoundry.sh/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup-zksync: installing foundry (version nightly, tag nightly)
foundryup-zksync: downloading latest forge, and cast
######################################################################################################################################################################################################################################################### 100.0%
foundryup-zksync: downloading manpages
######################################################################################################################################################################################################################################################### 100.0%
foundryup-zksync: installed - forge 0.0.2 (9093c3c 2024-10-31T00:33:32.561294000Z)
foundryup-zksync: installed - cast 0.0.2 (9093c3c 2024-10-31T00:33:32.617382000Z)
foundryup-zksync: done!

forge create --zksync --rpc-url https://api.testnet.abs.xyz contracts/src/GasliteDrop.sol:GasliteDrop --private-key $PK
[⠃] Using zksolc-1.5.4
[⠊] Compiling (zksync)
No files changed, compilation skipped
Deployer: 0xe06ad0Acb3956f98C9dF77Bd02cdFD4426b2208d
Deployed to: 0xF8289FD0DC7813babfe5c2948f02D40757b7CFF6
Transaction hash: 0x3513de6ff60f52c108c007ce0e0c02e7c1d426d9658f51a15b6e921c98bcf2f4

forge verify-contract --zksync \
--rpc-url https://api.testnet.abs.xyz \
--verifier-url https://api-explorer-verify.testnet.abs.xyz/contract_verification \
--verifier zksync \
0xF8289FD0DC7813babfe5c2948f02D40757b7CFF6 \
contracts/src/GasliteDrop.sol:GasliteDrop
Start verifying contract `0xF8289FD0DC7813babfe5c2948f02D40757b7CFF6` deployed on 11124

Submitting verification for [contracts/src/GasliteDrop.sol:GasliteDrop] at address 0xF8289FD0DC7813babfe5c2948f02D40757b7CFF6.
Verification submitted successfully. Verification ID: 3443
Checking verification status for ID: 3443 using verifier: zksync at URL: https://api-explorer-verify.testnet.abs.xyz/contract_verification
Error:
Verification error: Compilation error
```

**Comment by alhenaai** [comment]
It appears that you are encountering a compilation error during the verification process of your contract `GasliteDrop` on the Abstract Testnet. This issue might be related to the specific settings or versions used in your deployment and verification commands.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Compiler Versions:** Ensure that the `zkSolc` version used for deployment matches the one used during verification. Discrepancies in compiler versions can lead to bytecode mismatches.

2. **Review Contract Code:** Sometimes, subtle issues in the contract code can cause compilation errors during verification. Review your contract code for any potential issues that might not affect deployment but could impact compilation during verification.

3. **Manual Verification:** Try manually verifying the contract through the block explorer's UI if available. This can sometimes provide more detailed error messages or different results.

4. **Update Tools:** Ensure that all your tools (`foundry`, `forge`, etc.) are updated to their latest versions. Sometimes, bugs in earlier versions can cause unexpected issues during contract interactions.

5. **Consult Documentation:** Review the [ZKsync documentation](https://docs.zksync.io/) for any updates or notes on contract verification that might be relevant to your issue.

If these steps do not resolve the issue, you might want to reach out for more specific support or check if there are any known issues with the tools or network you are using. For more in-depth discussion and support, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by coffee-the-dev** [comment]
It seems like this may be compiler version related?

Changing the pragma to `^0.8.24` fixes the verification issues, but this shouldn't be required for verification.

Verified deployment here succesfully: https://explorer.testnet.abs.xyz/address/0xa28EBcbda3762A24ABF7BF72441290B592382e9B#contract

**Reply by bxpana** [reply]
I was able to replicate this on ZKsync Era and had the same things where it worked after changing it to `^0.8.24` I'll check with the compiler team and see if there is a cause for that.

**Reply by dutterbutter** [reply]
@coffeexcoin I see the issue is with the verification request from `foundry-zksync`. I am working on a fix. I was able to successfully verify via hardhat but will update here when the PR is merged! Thanks for posting sir 🙏

**Reply by bxpana** [reply]
I just submitted a [PR](https://github.com/PopPunkLLC/GasliteDrop/pull/43) to your repo that adds a `foundry.toml` file to use zksolc `1.5.6` that works with deploying and verifying.

**Reply by dutterbutter** [reply, marked as answer]
@coffeexcoin will be fixed once https://github.com/matter-labs/foundry-zksync/pull/690 is merged, release will automatically roll out tonight.

Again thanks for posting sir 🙏
